src/routes/vendor.tsx

```diff
@@ -33,11 +33,22 @@ export default function VendorPage() {
     },
   )
 
+  const { data: locationsData, isLoading: isLocationsLoading, error: locationsError } = useGetLocationsQuery()
+
   useQueryErrorHandler(error)
+  useQueryErrorHandler(locationsError)
 
   useEffect(() => {
-    setShowLoadingOverlay(!isSuccess)
-  }, [isSuccess, setShowLoadingOverlay])
+    setShowLoadingOverlay(!isSuccess || isLocationsLoading)
```

Comment:
we can change 
```setShowLoadingOverlay(!isSuccess || isLocationsLoading)``` 
to 
```setShowLoadingOverlay(!isSuccess && !isLoactionSuccess)```  

Just to make it uniform with the success from fetch vendor data